src/net/src/tls_tcp.rs

```diff
@@ -0,0 +1,685 @@
+                    )
+                })?;
+            }
+            (None, Some(cert)) => {
+                // this will just load the leaf certificate from the file
+                self.inner
+                    .set_certificate_file(cert, SslFiletype::PEM)
+                    .map_err(|e| {
+                        Error::new(
+                            ErrorKind::Other,
+                            format!("failed to load certificate file: {}", e),
+                        )
+                    })?;
+            }
+            (None, None) => {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    "no certificate file or certificate chain file provided",
+                ));
+            }
+        }
+
+        let inner = self.inner.build().into_context();
+
+        Ok(TlsTcpConnector { inner })
+    }
+
+    pub fn verify(mut self, mode: SslVerifyMode) -> Self {
+        self.inner.set_verify(mode);
+        self
+    }
+
+    /// Load trusted root certificates from a file.
+    ///
+    /// The file should contain a sequence of PEM-formatted CA certificates.
+    pub fn ca_file<P: AsRef<Path>>(mut self, file: P) -> Self {
+        self.ca_file = Some(file.as_ref().to_path_buf());
+        self
+    }
+
+    /// Load a leaf certificate from a file.
+    ///
+    /// This loads only a single PEM-formatted certificate from the file which
+    /// will be used as the leaf certifcate.
+    ///
+    /// Use `set_certificate_chain_file` to provide a complete certificate
+    /// chain. Use this with the `set_certifcate_chain_file` if the leaf
+    /// certifcate and remainder of the certificate chain are split across two
+    /// files.
+    pub fn certificate_file<P: AsRef<Path>>(mut self, file: P) -> Self {
+        self.certificate_file = Some(file.as_ref().to_path_buf());
+        self
+    }
+
+    /// Load a certificate chain from a file.
+    ///
+    /// The file should contain a sequence of PEM-formatted certificates. If
+    /// used without `set_certificate_file` the provided file must contain the
+    /// leaf certificate and the complete chain of certificates up to and
+    /// including the trusted root certificate. If used with
+    /// `set_certificate_file`, this file must not contain the leaf certifcate
+    /// and will be treated as the complete chain of certificates up to and
+    /// including the trusted root certificate.
+    pub fn certificate_chain_file<P: AsRef<Path>>(mut self, file: P) -> Self {
+        self.certificate_chain_file = Some(file.as_ref().to_path_buf());
+        self
+    }
+
+    /// Loads the private key from a PEM-formatted file.
+    pub fn private_key_file<P: AsRef<Path>>(mut self, file: P) -> Self {
+        self.private_key_file = Some(file.as_ref().to_path_buf());
+        self
+    }
+}
+
+// NOTE: these tests only work if there's a `test` folder within this crate that
```

Comment:
What about a cert that would be valid for another 68 years?
- Why 68? Just in case boringSSL has the same issue as openSSL https://groups.google.com/g/mailing.openssl.users/c/3kK_f0ywCZQ

(In case not obvious, I'm not really serious about this)

Reply:
Lol. Won't be our problem then =)

Will follow-up on this, would be nice to have these tests enabled.